src/types.ts

```diff
@@ -30,8 +30,8 @@ export interface Config {
 /// Request specific interfaces
 ///
 
-export interface IndexRequest {
-  uid: string
+export interface IndexOptions {
+  uid?: string
```

Comment:
Why `uid` is still in IndexOptions?
It's not an option from the POV of this SDK and should be filled in the first parameter. If the user mentions an `uid` in the `options` it will overwrite the first one filled in the first parameter -> that is not what we want/expect when we use an SDK.
It gives complexity to the SDK for nothing.
Providing an SDK is also providing a structured framework to the user with known and expected behavior 🙂

Reply:
But I might miss-understood the typescript 😇

Reply:
I understand you did that because you have a `uidAndPrimaryKey` in your tests, but we should change the way we do the tests according to the user usage 😉